From 1db115bb9ae7fd285cd86a2334efdbc64a7b25c4 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:26:57 -0400
Subject: [net] genetlink: remove genl_unregister_mc_group()

Message-id: <e825859db0d2cfd2938ea609be72ee4614f90b0c.1410528958.git.jbenc@redhat.com>
Patchwork-id: 93949
O-Subject: [RHEL7.1 net PATCH 15/23] genetlink: remove genl_unregister_mc_group()
Bugzilla: 1140656
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1140656
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit 06fb555a273dc8ef0d876f4e864ad11cfcea63e0
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Tue Nov 19 15:19:35 2013 +0100

    genetlink: remove genl_unregister_mc_group()

    There are no users of this API remaining, and we'll soon
    change group registration to be static (like ops are now)

    Signed-off-by: Johannes Berg <johannes.berg@intel.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/net/genetlink.h b/include/net/genetlink.h
index a1ede3c..15f61da 100644
--- a/include/net/genetlink.h
+++ b/include/net/genetlink.h
@@ -177,8 +177,6 @@ static inline int _genl_register_family_with_ops(struct genl_family *family,
 int genl_unregister_family(struct genl_family *family);
 int genl_register_mc_group(struct genl_family *family,
       struct genl_multicast_group *grp);
-void genl_unregister_mc_group(struct genl_family *family,
-         struct genl_multicast_group *grp);
 void genl_notify(struct sk_buff *skb, struct net *net, u32 portid,
    u32 group, struct nlmsghdr *nlh, gfp_t flags);
 
diff --git a/net/netlink/genetlink.c b/net/netlink/genetlink.c
index 2a9c182..7b8e3a3 100644
--- a/net/netlink/genetlink.c
+++ b/net/netlink/genetlink.c
@@ -265,29 +265,6 @@ static void __genl_unregister_mc_group(struct genl_family *family,
  grp->family = NULL;
 }
 
-/**
- * genl_unregister_mc_group - unregister a multicast group
- *
- * Unregisters the specified multicast group and notifies userspace
- * about it. All current listeners on the group are removed.
- *
- * Note: It is not necessary to unregister all multicast groups before
- *       unregistering the family, unregistering the family will cause
- *       all assigned multicast groups to be unregistered automatically.
- *
- * @family: Generic netlink family the group belongs to.
- * @grp: The group to unregister, must have been registered successfully
- *  previously.
- */
-void genl_unregister_mc_group(struct genl_family *family,
-         struct genl_multicast_group *grp)
-{
- genl_lock_all();
- __genl_unregister_mc_group(family, grp);
- genl_unlock_all();
-}
-EXPORT_SYMBOL(genl_unregister_mc_group);
-
 static void genl_unregister_mc_groups(struct genl_family *family)
 {
  struct genl_multicast_group *grp, *tmp;
-- 
1.7.1